pkg/tests/upgrade_tests/upgrade_test.go

```diff
@@ -635,3 +636,234 @@ func copyMetrics(metrics []prompb.TimeSeries) []prompb.TimeSeries {
 	}
 	return out
 }
+
+func TestPrereleaseExtensionUpgrade(t *testing.T) {
+	var err error
+	var version string
+	ctx := context.Background()
+
+	db, dbContainer, closer := startDB(t, ctx)
+	defer closer.Close()
+
+	defer testhelpers.StopContainer(ctx, dbContainer, false)
+
+	// This will build promscale image from current codebase.
+	buildPromscaleImageFromRepo(t)
+
+	// start promscale & test upgrade-prerelease-extensions as false
+	func() {
+		connectorImage := "timescale/promscale:latest"
+		databaseName := "postgres"
+		flags := []string{"-upgrade-prerelease-extensions", "false"}
+		connector, err := testhelpers.StartConnectorWithImage(ctx, connectorImage, *printLogs, flags, databaseName)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+		defer testhelpers.StopContainer(ctx, connector, *printLogs)
+
+		err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		if version != "2.0.0-rc4" {
+			t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension false")
+		}
+	}()
+
+	extVersion := "2.0.0-rc2"
+	dropAndCreateExt(t, ctx, extVersion)
+
+	db, err = pgx.Connect(ctx, testhelpers.PgConnectURL("postgres", testhelpers.Superuser))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = db.QueryRow(ctx, `SELECT extversion FROM pg_extension where extname='timescaledb'`).Scan(&version)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	db.Close(ctx)
+	if version != extVersion {
+		t.Fatal("failed to verify upgrade extension with -upgrade-prerelease-extension false")
+	}
+
```

Comment:
I would maybe run  --upgrade-prerelease-extensions = false once here on -rc2 to make sure that it does not upgrade to rc4. then run with true